File: iOS_Native/NotingMe/NotingMe/DetailViewController.m
//
//  DetailViewController.m
//  NotingMe
//
//  Created by Ashish Nigam on 11/14/14.
//  Copyright (c) 2014 AshishNigam. All rights reserved.
//

#import "DetailViewController.h"

@interface DetailViewController ()

@end

@implementation DetailViewController

#pragma mark - Managing the detail item

- (void)setDetailItem:(id)newDetailItem {
    if (_detailItem != newDetailItem) {
        _detailItem = newDetailItem;
            
        // Update the view.
        [self configureView];
    }
}

- (void)configureView {
    // Update the user interface for the detail item.
    if (self.detailItem) {
        self.noteTextView.text = [[self.detailItem valueForKey:@"noteText"] description];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self configureView];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)textViewDidBeginEditing:(UITextView *)textView
{
     NSLog(@"=============== mynote=============== textViewDidBeginEditing");
}
- (void)textViewDidEndEditing:(UITextView *)textView
{
    NSLog(@"=============== mynote=============== textViewDidEndEditing");
}

- (void)textViewDidChange:(UITextView *)textView
{
    NSLog(@"=============== mynote=============== textViewDidChange");

    [self.detailItem setValue:textView.text forKey:@"noteText"];
}

- (IBAction)textChangesDone:(id)sender {
    if ([self.noteTextView.text isEqualToString:@""]) {
        // move back
        [self.navigationController popToRootViewControllerAnimated:YES];
    }else{
        [self.noteTextView resignFirstResponder];
    }
}
@end

File: iOS_Native/SwiftBased_Project/MyPlaces/MyPlaces/DetailViewController.swift
//
//  DetailViewController.swift
//  MyPlaces
//
//  Created by Ashish Nigam on 12/01/16.
//  Copyright © 2016 Ashish. All rights reserved.
//

import UIKit

import SWFKit

class DetailViewController: UIViewController {

    @IBOutlet weak var detailDescriptionLabel: UILabel!
    
    var swfIns : SFMath = SFMath(newMessage: "ashish nigam init");
    var swfIns1 : SFMath = SFMath(newMessage: "ashish nigam init will print this as more than 20");
    var swfIns2 : SFCalculator = SFCalculator(newSFCalculator: "ashish nigam init will print this as more than 20");
    var swWave : SFSineWave = SFSineWave();

    var detailItem: AnyObject? {
        didSet {
            // Update the view.
            self.configureView()
        }
    }

    func configureView() {
        // Update the user interface for the detail item.
        if let detail = self.detailItem {
            if let label = self.detailDescriptionLabel {
                label.text = detail.valueForKey("timeStamp")!.description
            }
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.configureView()
        
        swfIns.printCurrentString("In view did load with framework");
        swfIns1.printCurrentString("\n In view did load with framework");
        swfIns2.printCurrentString("\n new calculator");
        swWave.printCurrentWave();
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

